homeassistant/components/xiaomi_miio/config_flow.py

```diff
@@ -84,13 +83,7 @@ async def async_step_init(
             ):
                 errors["base"] = "cloud_credentials_incomplete"
                 # trigger re-auth flow
-                self.hass.async_create_task(
-                    self.hass.config_entries.flow.async_init(
-                        DOMAIN,
-                        context={"source": SOURCE_REAUTH},
-                        data=self.config_entry.data,
-                    )
-                )
+                self.config_entry.async_start_reauth(self.hass)
```

Comment:
I think a few of these are bugfixes that should probably be moved to separate prs

Reply:
I have made prel PR's for the two cases.
I think the test fixes can be left in this PR.